### PR TITLE
JavaScript Linting Round 2: Add Eslint + Prettier Config Combo to Webpack

### PR DESCRIPTION
**Pull request by sghoweri**
TLDR: add Prettier to Bolt to automatically standardize some very basic code formatting in our JavaScript -- line breaks, quotes, and spaces basically.

With the [Prettier plugin for VS Code](https://github.com/prettier/prettier-vscode) + enabling the `editor.formatOnSave` config, this makes writing consistently formatted JavaScript that looks good and is easier to read so much easier!

<hr>

Functionally, this PR makes four "small" changes to Bolt:

1. Wire up [Prettier](https://prettier.io/) to be used in conjunction with our current eslint config 
2. Disable two existing eslint config rules we've already had enabled (`linebreak` and `quote` rules) in lieu of Prettier's much smarter handling of both of these (there's a few valid instances when we do need to use double quotes / should have code break to the next line to be more readable)
3. Update our Webpack build config to check / recheck our JavaScript's syntax every time we make a change
4. Slightly update the verbosity in Pattern Lab's `.boltrc` config so linting warnings are visible while doing local development (without too much extra noise getting added)

With these 4 changes, the rest of the files updated in this PR address the formatting warnings / errors that get thrown when running a full lint check of the code, without introducing any crazy functional changes in the process 😃

### CODE REVIEW

**Comment by sghoweri** [comment]
> 1	Why do we now have both prettier *and* eslint? There's now more complexity (which, to be clear, is a bad thing-- it's just that much harder to be aware of everything that's going on). But additionally, it means that just because you fix linting errors with Prettier in you IDE (yay!) doesn't mean you don't have linting errors (huh? boo!)

@remydenton as long as we configure things properly, eslint and prettier should be working together on reporting issues in our JS (via one single unified lint command)

## So why prettier? 

https://prettier.io/docs/en/why-prettier.html

> By far the biggest reason for adopting Prettier is to stop all the on-going debates over styles. It is generally accepted that having a common style guide is valuable for a project and team but getting there is a very painful and unrewarding process. People get very emotional around particular ways of writing code and nobody likes spending time writing and receiving nits.

So why choose the "Prettier style guide" over any other random style guide? Because Prettier is the only "style guide" that is fully automatic. Even if Prettier does not format all code 100% the way you'd like, it's worth the "sacrifice" given the unique benefits of Prettier, don't you think?

## What’s the difference between Prettier and Eslint?

https://prettier.io/docs/en/comparison.html

Linters have two categories of rules:
**Formatting rules**: eg:  [max-len](http://eslint.org/docs/rules/max-len) ,  [no-mixed-spaces-and-tabs](http://eslint.org/docs/rules/no-mixed-spaces-and-tabs) ,  [keyword-spacing](http://eslint.org/docs/rules/keyword-spacing) ,  [comma-style](http://eslint.org/docs/rules/comma-style) ...

Prettier alleviates the need for this whole category of rules! Prettier is going to reprint the entire program from scratch in a consistent way, so it's not possible for the programmer to make a mistake there anymore :)


**Code-quality rules**: eg  [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars) ,  [no-extra-bind](http://eslint.org/docs/rules/no-extra-bind) ,  [no-implicit-globals](http://eslint.org/docs/rules/no-implicit-globals) ,  [prefer-promise-reject-errors](http://eslint.org/docs/rules/prefer-promise-reject-errors) ...
Prettier does nothing to help with those kind of rules. They are also the most important ones provided by linters as they are likely to catch real bugs with your code!

^ that’s why eslint is typically used in conjunction with Prettier so you get the best of both worlds

**Comment by remydenton** [comment]
Ah, got it, they are _designed_ to run side-by-side: https://prettier.io/docs/en/eslint.html.  Thanks for bearing with me on the dumb question.

I'm still curious how best to test.  I do see a number of lint-like errors when running `npm run bootstrap`-- but then no apparent errors when I intentionally introduce issues, like removing the newline from the end of a file.

**Comment by sghoweri** [comment]
> I'm still curious how best to test. I do see a number of lint-like errors when running npm run bootstrap-- but then no apparent errors when I intentionally introduce issues, like removing the newline from the end of a file.

The two ways this should run (after initially running `npm run setup` to install the new deps) is either:

1. via the normal Webpack watch task / recompiles (`npm start`) in the pattern-lab folder which should lint all the JS files being watched and bundled 

2. by running the `npm run lint:js` or `npm run lint` commands in the root of the repo...

Can you try changing one of our component JS files to use double quotes instead of single quotes to see if that triggers something? Lmk if that doesn’t work for ya!

**Comment by remydenton** [comment]
Shoot, got some merge conflicts there-- can I get you to merge this in @sghoweri ?

**Comment by sghoweri** [comment]
@remydenton sure can 👍

I'll tackle the merge conflicts + getting the docs on IDE integration for VS Code, PHP Storm and Sublime in just a bit!
